ios/NebulaNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -116,6 +114,25 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
 //        completionHandler()
 //    }
     
+    private func findManager() async throws -> NETunnelProviderManager {
+        let targetProtoConfig = self.protocolConfiguration as? NETunnelProviderProtocol;
+        let targetID = targetProtoConfig?.providerConfiguration!["id"] as? String;
```

Comment:
Would be nice if we could avoid blowing up on `targetProtoConfig?.providerConfiguration!` but should be pretty obvious if we stop using our provided config with an `id` in favor of something else.

Reply:
I added some guards and a note in `site.swift` where we set the providerConfig and id.